CONTRIBUTING.md

```diff
@@ -0,0 +1,127 @@
+
+For example, the following will enable connecting your local development environment to Upbound Cloud:
+
+```
+export LOCALDEV_CONNECT_CP_ORG=<YOUR_UBC_ORG>
+export LOCALDEV_CONNECT_API_TOKEN=<YOUR_ACCESS_TOKEN>
+
+make build local-dev
+```
+
+### Cleanup
+
+To clean up local dev environment, first delete self hosted control plane (if connected) from Upbound Cloud Console
+and then run:
+
+```
+make local.down
+```
+
+### Validation
+
+To run validation tests locally, run:
+
+```
+make e2e.run
+```
+
+To run validation test including Upbound Cloud connectivity, run:
+
+```
+export LOCALDEV_CONNECT_CP_ORG=<YOUR_UBC_ORG>
+export LOCALDEV_CONNECT_API_TOKEN=<YOUR_ACCESS_TOKEN>
+
+make e2e.run
+```
+
+## Release Process
+
+A UXP release is cut by following the steps below:
+
+1. **branch repo**: Create a new release branch using the GitHub UI for the
+   repo.
+1. **release branch prep**: Make any release-specific updates on the release
+   branch.
+1. **tag release**: Run the `Tag` action on the _release branch_ with the
+   desired version (e.g. `v0.14.0`).
+1. **build/publish**: Run the `CI` action on the release branch with the version
+   that was just tagged.
+1. **tag next pre-release**: Run the `tag` action on the main development branch
+   with the `rc.0` for the next release (e.g. `v1.3.0-up.1-rc.1`).
+1. **verify**: Verify all artifacts have been published successfully, perform
+   sanity testing.
+1. **promote**: Run the `Promote` action to promote release to desired
+   channel(s).
+1. **release notes**: Publish well authored and complete release notes on
+   GitHub.
+1. **announce**: Announce the release on Twitter, Slack, etc.
+
+After a release is cut, we need to publish it to the following distribution channels.
+
+### Operator Lifecycle Manager
+
+OLM uses its own packaging format for its marketplaces.
+
+1. **download bundle**: Download the bundle from the following link:
+   ```
+   VERSION=<release version>
+   curl -sL https://releases.upbound.io/universal-crossplane/stable/$VERSION/olm/$VERSION.tar.gz  | tar xz
+   ```
+1. **open pull requests**: Extract the content into a new folder named `$VERSION`
+   and open the PRs to the following directories:
+    * https://github.com/operator-framework/community-operators/tree/master/community-operators/universal-crossplane
+    * https://github.com/operator-framework/community-operators/tree/master/upstream-community-operators/universal-crossplane
+1. **sanity check**: Once the PRs are merged and CI in main branch completes the
+   propagation, you should see the new version in [OperatorHub product page](https://operatorhub.io/operator/universal-crossplane).
+
+### AWS Marketplace
+
+AWS Marketplace installations are done using usual Helm commands but the images
+need to exist in given ECR repositories. Currently, due to technical limitations,
+it is not possible to define IAM policies on the allocated marketplace ECR repository
```

Comment:
We're still waiting to hear from them. The repositories we're given do not show up in AWS console.

Reply:
I have [figured it](https://github.com/upbound/universal-crossplane/issues/48#issuecomment-863655798) out but I won't be able to implement it on time. So, let's keep that as is and change it once the automation is implemented.